src/main/java/com/flagsmith/config/FlagsmithCacheConfig.java

```diff
@@ -151,18 +156,18 @@ public FlagsmithCacheConfig build() {
     }
   }
 
-  class FlagsmithInternalCache implements FlagsmithCache {
+  public class FlagsmithCacheImpl implements FlagsmithCache {
 
-    private final Cache<String, FlagsAndTraits> cache;
+    private final Cache<String, List<FeatureStateModel>> cache;
```

Comment:
I'm not sure that `FeatureStateModel` is the correct thing to use for the cache interface is it? Shouldn't it be `Flag`?